velox/type/DecimalUtil.h

```diff
@@ -301,6 +302,79 @@ class DecimalUtil {
     return remainder * resultSign;
   }
 
+  /// Returns the max required size to convert the decimal of this precision and
+  /// scale to varchar. A varchar's size is estimated with unscaled value
+  /// digits, dot, leading zero, and possible minus sign.
+  static int32_t maxStringViewSize(int precision, int scale) {
+    int32_t rowSize = precision + 1; // Number and symbol.
+    if (scale > 0) {
+      ++rowSize; // A dot.
+    }
+    if (precision == scale) {
+      ++rowSize; // Leading zero.
+    }
+    return rowSize;
+  }
+
+  /// @brief Convert the unscaled value of a decimal to varchar and write to raw
+  /// string buffer from start position.
+  /// @tparam T The type of input value.
+  /// @param unscaledValue The input unscaled value.
+  /// @param scale The scale of decimal.
+  /// @param maxVarcharSize The estimated max size of a varchar.
+  /// @param startPosition The start position to write from.
+  /// @return A string view.
+  template <typename T>
+  static StringView convertToStringView(
```

Comment:
Any particular reason to return a StringView here. It seems std::string_view would be more appropriate. 

convertToStringView names doesn't seem quite accurate. Perhaps, castToString is a better name especially since error messages in the implementation are using 'cast' term.

Reply:
I just move the function. 
I suggest to return `actual size` instead of `StringView` in original  PR but not accepted, then the caller could wrap `StringView(startPosition, returnSize)`.

I think it is because the caller needs `StringView`.
Can I return `size`? The pointer is the input argument `char* const startPosition`

Reply:
Returning size would be fine.

> I just move the function.

True, but you are moving "private" code contained within a module into a "public" space where it can be used by many modules. Public APIs need to be designed carefully as they are harder to change.